src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/core/stats/collectors/aggregator/SampleAggregator.java

```diff
@@ -0,0 +1,168 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.collectors.aggregator;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.collectors.Collector;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.eval.Statistics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.eval.impl.Count;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.eval.impl.Max;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.eval.impl.Mean;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.eval.impl.Min;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.eval.impl.NamedCounter;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.eval.impl.Sample;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.eval.impl.StatisticImpl;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.eval.impl.Sum;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.eval.impl.vals.Value;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.format.Formatter;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.stats.measurements.MeasurementSet;
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * This class is mainly to collect stats between runs of the RCA framework before we can write them
+ * using the Stats Collector.
+ *
+ * <p>This is suitable for cases where we want to calculate statistics before we report it, e.g the
+ * RCA graph evaluation. We want to know the long pole in the Graph node execution and how much it
+ * deviates from the mean but we also don't want to report the time taken by each graph node.
+ */
+public class SampleAggregator implements Collector {
```

Comment:
We can call this StatsAggregator. Sample is used as a statistic as well and can be confusing.

Reply:
Its goes not aggregates the stats but calculates them from the samples, though ? I am open to better names. Let me know what would sound more intuitive